File: 2/Hardware.cpp
#include "Product.cpp"

class Hardware:public Product
{
	private:
		string brand;
		string model;

	public:
		// construct with optional args
		Hardware(string brand, string model, int _price=0, string _idProduct="NULL")
		: Product(_price, _idProduct)
		{
			this->brand = brand;
			this->model = model;
		}

		// default construct
		Hardware(){}

		void setBrand(string brand)
		{
			this->brand = brand;
		}

		string getBrand()
		{
			return this->brand;
		}

		void setModel(string model)
		{
			this->model = model;
		}

		string getModel()
		{
			return this->model;
		}

		// deconstruct
		~Hardware(){}

		void printHardware()
		{
			cout << "Brand : " << this->brand << endl;
			cout << "Model : " << this->model << endl;
		}
};

File: 2/Hardware.php
<?php declare(strict_types=1);

include "Product.php";

class Hardware extends Product
{
	private string $brand;
	private string $model;

	// construct with optional args
	public function __construct(string $brand, string $model, int $price=0, string $idProduct="NULL")
	{
		parent::__construct($price, $idProduct);
		$this->brand = $brand;
		$this->model = $model;
	}

	public function setBrand(string $brand)
	{
		$this->brand = $brand;
	}

	public function getBrand()
	{
		return $this->brand;
	}

	public function setModel(string $model)
	{
		$this->model = $model;
	}

	public function getModel()
	{
		$this->model = $model;
	}

	public function printHardware()
	{
		echo "Brand : " . $this->brand . "<br>";
		echo "Model : " . $this->model . "<br>";
	}
}

?>